### PR TITLE
feat(deployer): send deployment archive to the builder

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description of change

Send the deployment archive to the builder.

## How has this been tested? (if applicable)

On staging and locally.

### CODE REVIEW

File: builder/src/lib.rs [hunk 1]
```diff
@@ -56,40 +56,70 @@ impl Service {
...
+                        Ok(None) => stdout_ended = true,
+                        Err(err) => {
+                            error!(deployment_id, "unexpected stdout stream close: {}", err);
+                            stdout_ended = true;
+                        }
+                    }
+                }
             }
-        });
+        }
```
Comment (oddgrd) [inline builder/src/lib.rs:110]: So this is to combine stderr and stdout, right? Could we use the same strategy we use in the [old builder](https://github.com/shuttle-hq/shuttle/blob/main/service/src/builder.rs#L248-L266)?

File: common-tests/src/builder.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    let builder_addr = SocketAddr::new(Ipv4Addr::LOCALHOST.into(), pick_unused_port().unwrap());
+    let builder_uri = format!("http://{}", builder_addr);
+    tokio::spawn(async move {
+        Server::builder()
+            .add_service(BuilderServer::new(builder))
+            .serve(builder_addr)
+            .await
+    });
+
+    // Wait for the logger server to start before creating a client.
```
Comment (oddgrd) [inline common-tests/src/builder.rs:31]:
```suggestion
    // Wait for the builder server to start before creating a client.
```

File: common-tests/src/builder.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    tokio::time::sleep(Duration::from_millis(200)).await;
+
+    let channel = Endpoint::try_from(builder_uri.to_string())
+        .unwrap()
+        .connect()
+        .await
+        .expect("failed to connect to logger");
```
Comment (oddgrd) [inline common-tests/src/builder.rs:38]:
```suggestion
        .expect("failed to connect to builder");
```

File: deployer/src/deployment/state_change_layer.rs [hunk 4]
```diff
@@ -269,6 +270,20 @@ mod tests {
...
+    impl Builder for RecorderMock {
+        async fn build(
+            &self,
+            _request: tonic::Request<BuildRequest>,
+        ) -> Result<tonic::Response<BuildResponse>, tonic::Status> {
+            Ok(Response::new(BuildResponse {
+                image: Vec::new(),
+                is_wasm: false,
+                secrets: Vec::new(),
+            }))
```
Comment (oddgrd) [inline deployer/src/deployment/state_change_layer.rs:283]:
```suggestion
            Ok(Response::new(BuildResponse::default())
```

**Review by oddgrd — COMMENTED** [review]
Looks good! Just left some nit/cleanup comments

File: builder/src/lib.rs (hunk 1 above)
Comment (iulianbarbu) [reply builder/src/lib.rs:110]: Open a ticket for this, will cover it in a next PR.

**Review by orhun — APPROVED** [review]

File: Makefile [hunk 5]
```diff
@@ -57,7 +57,7 @@ DD_ENV=production
 # make sure we only ever go to production with `--tls=enable`
 USE_TLS=enable
 CARGO_PROFILE=release
-RUST_LOG?=shuttle=debug,info
+RUST_LOG?=nbuild-core=warn,shuttle=debug,info
```
Comment (oddgrd) [inline Makefile:60]: Could we achieve this by passing a layer with an envfilter to the `setup_tracing` in builder?

File: .circleci/config.yml [hunk 6]
```diff
@@ -724,19 +724,19 @@ workflows:
               only: production
       - approve-push-unstable:
           type: approval
-          filters:
-            branches:
-              only: main
+          # filters:
+          #   branches:
+          #     only: main
```
Comment (oddgrd) [inline .circleci/config.yml:729]: We should remember to remove these before merge

File: Makefile (hunk 5 above)
Comment (iulianbarbu) [reply Makefile]: Done here: [2edaf12](https://github.com/shuttle-hq/shuttle/pull/1253/commits/2edaf12c10897b00a60881730c3e9259466df522)

File: .circleci/config.yml (hunk 6 above)
Comment (iulianbarbu) [reply .circleci/config.yml]: Done here: [350c959](https://github.com/shuttle-hq/shuttle/pull/1253/commits/350c959250e1e95014ab217f79ea457b82fc0442)

File: builder/src/main.rs [hunk 7]
```diff
@@ -17,7 +17,11 @@ use tracing::trace;
 async fn main() {
     let args = Args::parse();
 
-    setup_tracing(tracing_subscriber::registry(), Backend::Builder);
+    setup_tracing(
+        tracing_subscriber::registry(),
+        Backend::Builder,
+        Some("nbuild_core=warn,info"),
+    );
```
Comment (oddgrd) [inline builder/src/main.rs:24]: Did you try doing it like this? 
````rust
setup_tracing(
        tracing_subscriber::registry().with(EnvFilter::from_env("RUST_LOG=...")),
        Backend::Builder,
    );

**Pull request merged**